tests/nested/extra-snaps-assertions/task.yaml

```diff
@@ -0,0 +1,75 @@
+summary: create ubuntu-core image and execute the suite in a nested qemu instance
+
+systems: [ubuntu-16.04-64]
+
+prepare: |
+    . $TESTSLIB/prepare.sh
+    prepare_classic
+    prepare_each_classic
+
+    # determine arch related vars
+    case "$NESTED_ARCH" in
+    amd64)
+        QEMU="$(which qemu-system-x86_64)"
+        ;;
+    i386)
+        QEMU="$(which qemu-system-i386)"
+        ;;
+    *)
+        echo "unsupported architecture"
+        exit 1
+        ;;
+    esac
+
+    # create ubuntu-core image
+    mkdir -p /tmp/work-dir
+    snap install --devmode --beta ubuntu-image
+    # bind mount the snap binary installed from the package built with the branch to the binary used by ubuntu-image
+    mount -o bind /usr/bin/snap /snap/ubuntu-image/current/bin/snap
+
+    snap download core
+
+    /snap/bin/ubuntu-image --image-size 3G $TESTSLIB/assertions/nested-${NESTED_ARCH}.model --channel $CORE_CHANNEL --output ubuntu-core.img --extra-snaps core_*.snap
+    mv ubuntu-core.img /tmp/work-dir
+
+    # create assertion block device
+    genisoimage -volid cidata -joliet -rock -o assertions.disk $TESTSLIB/assertions/auto-import.assert
+
+    . $TESTSLIB/systemd.sh
+    systemd_create_and_start_unit nested-vm "${QEMU} -m 1024 -nographic -net nic,model=virtio -net user,hostfwd=tcp::8022-:22 -drive file=/tmp/work-dir/ubuntu-core.img,if=virtio,cache=none -drive file=${PWD}/assertions.disk,if=virtio,cache=none"
```

Comment:
Yes, I haven't seen any memory problems so far. Also keep in mind that the nested suite is marked as manual as a whole, we need to execute this test explicitly, maybe in response to some event (merges to master, new core published, etc) or as part of the nightly execution.